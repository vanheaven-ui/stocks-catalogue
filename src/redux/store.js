import { compose, createStore } from 'redux';
import rootReducer from './reducers';

const storeEnhancers = compose(
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(), // eslint-disable-line
);

const store = createStore(rootReducer, storeEnhancers);

export default store;
