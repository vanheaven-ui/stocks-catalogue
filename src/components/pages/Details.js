import { useParams } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import Loading from '../Loading';
import StockDetails from '../../containers/StockDetails';
import { getProfile, getQuotas } from '../../redux/actions';
import { profileEndPoint } from '../../constants';
import useFetch1 from '../../common/hooks/useFetch1';

const Details = () => {
  const dispatch = useDispatch();
  const { symbol } = useParams();
  const id = symbol;
  const profileURL = `${profileEndPoint}/${id}?apikey=${process.env.REACT_APP_STOCKS_API_KEY}`;
  const qURL = `${process.env.REACT_APP_QUOTA_API_URL}${id}?apikey=${process.env.REACT_APP_STOCKS_API_KEY}`;

  const { data: profile, isLoading } = useFetch1(profileURL);
  const { data: quota } = useFetch1(qURL);

  if ((profile.length > 0) && (quota.length > 0)) {
    dispatch(getProfile(profile));
    dispatch(getQuotas(quota));
  }
  return (
    <>
      { isLoading && <Loading color="blue" /> }
      { !isLoading && <StockDetails /> }
    </>
  );
};

export default Details;
