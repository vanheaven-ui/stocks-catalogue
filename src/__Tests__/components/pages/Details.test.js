import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Details from '../../../components/pages/Details';
import store from '../../../redux/store';
import rootNode from '../../helpers';

describe('Details page', () => {
  test('renders without crashing', () => {
    ReactDOM.render(<Provider store={store}><Details /></Provider>, rootNode());
  });

  test('doesnot change unexpectedly', () => {
    const DetailsTree = renderer.create(<Provider store={store}><Details /></Provider>).toJSON();
    expect(DetailsTree).toMatchSnapshot();
  });
});
