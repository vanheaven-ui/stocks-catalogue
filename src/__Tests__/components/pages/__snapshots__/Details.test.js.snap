// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page doesnot change unexpectedly 1`] = `
<section
  className="stockDetails"
  style={
    Object {
      "color": "#fff",
    }
  }
>
  <div
    className="container"
  />
</section>
`;
